pageserver/src/tenant.rs

```diff
@@ -3209,6 +3213,51 @@ fn run_initdb(
     Ok(())
 }
 
+/// Create special neon_superuser role, that's a slightly nerfed version of a real superuser
+/// that we give to customers
+fn create_neon_superuser(
+    conf: &'static PageServerConf,
+    target_dir: &Path,
+    pg_version: u32,
+) -> anyhow::Result<()> {
+    let createuser_bin_path = conf.pg_bin_dir(pg_version)?.join("createuser");
```

Comment:
`createuser` binary just connects to postgres and executes `CREATE ROLE`, but after initdb there is no postgres running. Hence you have tests erroring out with `InternalServerError(createuser failed: \'createuser: error: connection to server on socket "/tmp/.s.PGSQL.5432" failed: No such file or directory\n'`.

I think we have to do that in the compute_ctl

Reply:
We also have to think about backpatching existing timeline, as this code is only executed during tenant initialization and we will need something that will be executed on every (almost) compute start

As we only need `neon_superuser` when user tries to create a new role, we can probably put this backward compatibility code with `neon_superuser` creation at the top of `apply_config`

Reply:
All very good points, thank you. I've moved it to the top of apply_config.